notifications/migrations/0003_notification_data.py

```diff
@@ -13,7 +12,7 @@ class Migration(migrations.Migration):
         migrations.AddField(
             model_name='notification',
             name='data',
-            field=jsonfield.fields.JSONField(null=True, blank=True),
+            field=models.JSONField(null=True, blank=True),
```

Comment:
I think this was a text column type, so changing this here may break existing deployments using postgresql as in my understanding postgres has a native type for this.

Therefore, in order to maintain backward compatibility, I believe we should change this to a simple TextField, as the DB column behind the scenes is the same and it should not cause issues, but we'll have to triple check.

This is somewhat similar to what @emiraldo has commented before me, but I think we should be able to get away with fewer steps.

@emiraldo what do you think of the above?

Reply:
Yes "jsonfield" is using models.TextField. Django models.JSONField type on db level depends on exact db. For example, Postgres, it will be jsonb type https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.JSONField
Previously, I've done migration on my app with Postgres Django and django-notification. "data" field was altered to jsonb on db level and everything was fine.
I can agree with @emiraldo that direct ALTER FIELD might be potentially dangerous, in case of server fail during the process.

As far as know, changing already applied migration (our case 0003_notification_data.py) file will not push Django to reapply this migration, because Django remember this migration in its table. https://realpython.com/digging-deeper-into-migrations/

Reply:
You're right, it won't cause issues to existing deployments.
On Postgresql, by default all statements are run in transactions and are atomic, if the transaction fails, the changes are rolledback, so I am not worried about that.

Reply:
We could test applying this migration to a notifications_notification table with data and see what happens 🤔

Reply:
Guys, just to sum up..
1) Switching from "jsonfield" models.TextField to models.JSONField in this PR is acceptable? Or you want only to switch to native Django TextFeild?
2) Migrations.. We could just alter field and rely on db transaction or it would be better to work with field duplication as @emiraldo mentioned? Postgres and SQLite have DDL transactions while MySQL and Oracle have not.
3) Test. 
Just create a model with json compatible TextField with add field migration file. 
Then change this migration file, make a new migration and run migrate?

Reply:
I think that the support to be able to do searches with the ORM when using a JSONField gives a lot of value to the library, because it gives freedom of use and makes less complicated the searches because you don't need to do ‘icontains’ every time you need to search information from the ‘data’ field.

The problem of support for other databases is a concern because it closes the public that can use the library, although the use of USE_JSONFIELD could have a specification (in the documentation) that warns that the support is as specified by the version of django that is being used.